src/components/edit-movie/edit-movie.tsx

```diff
@@ -2,30 +2,81 @@ import classNames from 'classnames';
 import { AppButton } from 'components/app-button/app-button';
 import { Genres } from 'components/menu/menu';
 import { Modal } from 'components/modal/modal';
+import { useState } from 'react';
+import { useAppDispatch } from 'redux/hooks';
+import { addMovieAction, updateMovieAction } from 'redux/movies';
+import { Movie } from 'types/movie';
 import styles from './edit-movie.module.scss';
 
 const genres = [Genres.Comedy, Genres.Crime, Genres.Documentary, Genres.Horror];
 
+type MovieFormValues = Partial<Movie>;
+
+export enum Actions {
+  Add = 'add',
+  Edit = 'edit',
+}
+
+enum MovieFields {
+  Title = 'title',
+  ReleaseDate = 'release_date',
+  PosterPath = 'poster_path',
+  Genres = 'genres',
+  VoteAverage = 'vote_average',
+  Runtime = 'runtime',
+  Overview = 'overview',
+}
+
 interface Props {
   title: string;
+  movie?: Movie;
   onClose: () => void;
+  action: Actions;
 }
 
-export const EditMovie = ({ title, onClose }: Props) => {
+export const EditMovie = ({ title, movie, onClose, action }: Props) => {
+  const dispatch = useAppDispatch();
+
+  const [movieData, setMovieData] = useState<MovieFormValues | undefined>(
+    movie,
+  );
+
   const handleSubmit = () => {
+    if (!movieData || !isValidMovie(movieData)) {
+      return;
+    }
+
+    if (action === Actions.Add) {
+      dispatch(addMovieAction(movieData));
+    }
+    if (movie?.id) {
+      dispatch(updateMovieAction({ ...movieData, id: movie.id }));
+    }
     onClose();
   };
   const handleReset = () => {
     onClose();
   };
 
+  const setData = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+    field: string,
+  ) => {
+    setMovieData((prevState) => ({
+      ...prevState,
+      [field]: e.target.value,
+    }));
+  };
+
   return (
     <Modal onClose={handleReset}>
       <div className={styles.title}>{title}</div>
       <div className={styles.content}>
         <div className={styles.textWrapper}>
           <label className={styles.label}>Title</label>
           <input
+            value={movieData?.[MovieFields.Title]}
+            onChange={(e) => setData(e, MovieFields.Title)}
```

Comment:
My bad, I just noticed: you can use `name` attribute for determining the name of your field. In this case, you can get this name via `e.target.value`. It will simplify the logic and binding to method like:
```suggestion
            onChange={setData}
```

Let's make it a little better :)